### PR TITLE
remove weave{proxy,plugin} remnants

### LINKED ISSUE #3035: some weave{plugin,proxy} remnants are left
```
./tools/integration/config.sh:118:        PLUGIN_ID=$(docker_on "$host" ps -aq --filter=name=weaveplugin)
./test/config.sh:139:        for C in weaveplugin weave ; do
./test/config.sh:325:        PLUGIN_FILTER=$(docker_on $host inspect -f 'grep -v {{printf "%.12s" .Id}}' weaveplugin 2>/dev/null) || PLUGIN_FILTER=cat
./test/090_docker_restart_policy_2_test.sh:15:assert "docker_on $HOST1 inspect -f '{{.HostConfig.RestartPolicy.Name}}' weave weaveproxy" "no"
./site/weave-docker-api/troubleshooting-proxy.md:22:weaveproxy container logs using:
./site/weave-docker-api/troubleshooting-proxy.md:24:    docker logs weaveproxy
```

### PR DESCRIPTION
**Pull request by rade**
Fixes #3035.

### CODE REVIEW

**Comment by rade** [comment]
The 2.0 branch CI [fails](https://circleci.com/gh/weaveworks/weave/9284) with exactly the same error as this branch. @bboreham @marccarre HELP!

File: test/090_docker_restart_policy_2_test.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ assert_raises "check_restart $HOST1 weave"
 # in the old containers being removed and new ones created
 weave_on $HOST1 stop
 weave_on $HOST1 launch --no-restart
-assert "docker_on $HOST1 inspect -f '{{.HostConfig.RestartPolicy.Name}}' weave weaveproxy" "no"
+assert "docker_on $HOST1 inspect -f '{{.HostConfig.RestartPolicy.Name}}' weave" "no"
```
Comment (bboreham) [inline test/090_docker_restart_policy_2_test.sh]: This was a bug anyway - when we did have a `weaveproxy` container the result should have been
`no`
`no`

**Comment by bboreham** [comment]
CI failure fixed in 01194ee267ea38b2750b4134a0a782c963e8e95d - `nginx` image was missing the `ping` command which the test relied on.
Evidently this changed in the last two days, but I could not trace the exact trigger.

Using the `weaveworks:nettest` image is better anyway, as it is already on the machine and we can use its built-in network test too (left for later).

**Comment by rade** [comment]
This is failing CI with
```
--- FAIL: TestMain (0.00s)

panic: Specify weave args after --
```
I suspect the cause is other changes dragged in through the build-tools update, in particular weaveworks/build-tools#103.

**Comment by rade** [comment]
The failure happens because of [a recent build-tools change](https://github.com/weaveworks/build-tools/compare/80dd78e6e35e2b5d24960a8b128675cfe22707d4...b8b4d64c18bfcbbaede25cd67aa28eb281a26ca5#diff-098f6bcd4621d373cade4e832627b4f6R81). That was introduced in weaveworks/build-tools#103, apparently to [make tests in scope under `prog` actually execute](https://github.com/weaveworks/scope/pull/2567/files#diff-90a921ac5d7b373b0101b1f554146104R77).

**Pull request merged**